cluster/swarm/cluster.go

```diff
@@ -220,8 +220,12 @@ func (c *Cluster) createContainer(config *cluster.ContainerConfig, name string,
 
 	c.scheduler.Unlock()
 
-	log.WithFields(log.Fields{"NodeName": n.Name, "NodeID": n.ID}).Debugf("Scheduling container %s to ", name)
 	container, err := engine.CreateContainer(config, name, true, authConfig)
+	containerFlag := name
+	if containerFlag == "" && err == nil {
+		containerFlag = stringid.TruncateID(container.ID)
+	}
+	log.WithFields(log.Fields{"NodeName": n.Name, "NodeID": n.ID}).Debugf("Scheduling container %s to ", containerFlag)
```

Comment:
Before engine start to create container, swarm has already scheduled container to some node. If scheduler has some error, it will return before creating container by engine.

    nodes, err := c.scheduler.SelectNodesForContainer(c.listNodes(), config)

	if withImageAffinity {
		config.RemoveAffinity("image==" + config.Image)
	}

	if err != nil {
		c.scheduler.Unlock()
		return nil, err
	}

Then, if engine create container occur some error, no matter what kind of this error ,we'd better log scheduler info . If error occurs, we can find which node this container is going to live. Find that node and see what happens in that node.

@allencloud